### PR TITLE
Allow purple connections to set no_proxy per account.

### PR DESCRIPTION
**Pull request by edmcman**

### CODE REVIEW

File: protocols/purple/purple.c [hunk 1]
```diff
@@ -364,6 +368,14 @@ static void purple_sync_settings(account_t *acc, PurpleAccount *pa)
 		const char *name = "line-auth-token";
 		purple_account_set_string(pa, name, set_getstr(&acc->set, name));
 	}
+
+        PurpleProxyInfo *proxyinfo = purple_account_get_proxy_info(pa);
+        if (!proxyinfo) {
+          proxyinfo = purple_proxy_info_new();
+        }
+        assert (proxyinfo);
```
Comment (dequis) [inline protocols/purple/purple.c:376]: We don't normally use assert (with break), usually just `g_return_if_fail` or similar. What are you asserting here anyway? If it's out of memory then just don't.

Comment (edmcman) [reply protocols/purple/purple.c]: Thanks, I figured that wasn't the right thing to do, but wasn't sure what to do instead.

Comment (Thaodan) [reply protocols/purple/purple.c]: Squash that with the previous commit?

Comment (dequis) [reply protocols/purple/purple.c]: I normally just do that on merge

Comment (Thaodan) [reply protocols/purple/purple.c]: What about merging this?

**Review by jelmer — CHANGES_REQUESTED** [review]
Sorry for the delay; can you rebase this so we can get the benefit of GH actions?

**Comment by edmcman** [comment]
I'll do this but I'm on vacation this week.

**Comment by edmcman** [comment]
I have rebased this to the current master.  It looks like someone has to approve the GH actions.

**Comment by jelmer** [comment]
> I have rebased this to the current master. It looks like someone has to approve the GH actions.

Just did so.

**Comment by edmcman** [comment]
@jelmer Is there anything further?

**Pull request merged**